### PR TITLE
Fix missing HybridSearch in copy method of SearchOptions

### PR DESCRIPTION
**Pull request by nielsheeren**
The query parameters to configure hybrid search behaviors are not send to the API because they are not copied when cloning the search options. 

---
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by ericvdberge** [comment]
+1 I have this bug and need a fix as well.

**Comment by ShivangiReja** [comment]
@nielsheeren Thank you so much for your contribution in identifying and fixing the issue! Your help is greatly appreciated, and it's contributors like you who make our SDKs better.

File: sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs [hunk 1]
```diff
@@ -428,6 +428,7 @@ private static void Copy(SearchOptions source, SearchOptions destination)
             destination.QuerySpeller = source.QuerySpeller;
             destination.SemanticSearch = source.SemanticSearch;
             destination.VectorSearch = source.VectorSearch;
+            destination.HybridSearch  = source.HybridSearch;
```
Comment (ShivangiReja) [inline sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs:431]: Could you please update [this](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/search/Azure.Search.Documents/tests/DocumentOperations/SearchTests.cs#L1006) test case and add `HybridSearch` property for verification? Thank you!

Comment (nielsheeren) [reply sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs:431]: @ericvdberge Added the unit test

**Comment by nielsheeren** [comment]
> @microsoft-github-policy-service agree

@microsoft-github-policy-service agree

**Review by ShivangiReja — APPROVED** [review]

**Pull request merged**